cadc-storage-permissions/src/main/java/org/opencadc/inventory/permissions/PermissionsClient.java

```diff
@@ -89,17 +88,33 @@ public PermissionsClient() {
     }
     
     /**
-     * Get the permissions information about the file identified by fileURI.
+     * Get the read permissions information about the file identified by fileURI.
      * 
      * @param artifactURI Identifies the artifact for which to retrieve grant information.
-     * @return The grant information.
+     * @return The read grant information.
      * 
      * @throws ResourceNotFoundException If the file could not be found.
      * @throws TransientException If an unexpected, temporary exception occurred. 
      */
-    public Grant get(URI artifactURI) throws ResourceNotFoundException, TransientException {
+    public ReadGrant getReadGrant(URI artifactURI)
+        throws ResourceNotFoundException, TransientException {
         // TODO
-        return new Grant(artifactURI);
+        return new ReadGrant(artifactURI);
     }
-    
+
+    /**
+     * Get the read-write permissions information about the file identified by fileURI.
+     *
+     * @param artifactURI Identifies the artifact for which to retrieve grant information.
+     * @return The read-write grant information.
+     *
+     * @throws ResourceNotFoundException If the file could not be found.
+     * @throws TransientException If an unexpected, temporary exception occurred.
+     */
+    public ReadWriteGrant getReadWriteGrant(URI artifactURI)
```

Comment:
Agree with other review that this should, be getWriteGrant. The read-only vs read-write is a concept in the way we maintain permissions (read-write means "this group has two permissions")... but here these methods are about what action is being authlorised.

Reply:
Renamed.